### PR TITLE
Add initialization for first state in solver

### PR DESCRIPTION
**Pull request by edantec**
dx0 should be equal to the difference between x0 constraint as defined in the problem, and x0 variable as provided in the guess.

### CODE REVIEW

File: src/core/solvers/ddp.cpp [hunk 1]
```diff
@@ -246,8 +246,13 @@ void SolverDDP::forwardPass(const double& steplength) {
   for (std::size_t t = 0; t < T; ++t) {
     const boost::shared_ptr<ActionModelAbstract>& m = problem_->get_runningModels()[t];
     const boost::shared_ptr<ActionDataAbstract>& d = problem_->get_runningDatas()[t];
-
-    m->get_state()->diff(xs_[t], xs_try_[t], dx_[t]);
+    
+    if (t == 0) {
+      m->get_state()->diff(problem_->get_x0(), xs_[0], dx_[0]);
```
Comment (proyan) [inline src/core/solvers/ddp.cpp]: I think it is better if you don't specialize for t==0 here. Rather, you should look at 
1) https://github.com/loco-3d/crocoddyl/blob/devel/src/core/solvers/ddp.cpp#L43
2) https://github.com/loco-3d/crocoddyl/blob/devel/src/core/solvers/ddp.cpp#L155
3) https://github.com/loco-3d/crocoddyl/blob/devel/src/core/solvers/ddp.cpp#L345
At all these places, we are assigining xs_try_ to be equal to get_x0(). Instead, we need to make sure that dx0 is properly accounted for in these assignments.

File: src/core/solvers/fddp.cpp [hunk 2]
```diff
@@ -179,7 +179,12 @@ void SolverFDDP::forwardPass(const double& steplength) {
     } else {
       m->get_state()->integrate(xnext_, fs_[t] * (steplength - 1), xs_try_[t]);
     }
-    m->get_state()->diff(xs_[t], xs_try_[t], dx_[t]);
+    if (t == 0) {
```
Comment (proyan) [inline src/core/solvers/fddp.cpp]: same here.

**Comment by cmastalli** [comment]
We are already properly handling the initial conditions in both solvers. Therefore, we don't need this PR (and most probably there is an error).

See my explanation here: **https://github.com/loco-3d/crocoddyl/issues/728#issuecomment-605958175**

Let me know if you agree.

**Comment by edantec** [comment]
Ok, I agree with that: I didn't see that initial condition was already handled.

**Pull request closed without merging**